raiden_libs/test/mocks/client.py

```diff
@@ -288,6 +305,13 @@ def request_paths(self, target_address: Address, **kwargs) -> PathsRequest:
         request.signature = encode_hex(sign_data(self.privkey, request.serialize_bin()))
         return request
 
+    @staticmethod
+    def on_paths_reply_message(self, pfs_reply: PathsReply):
+        """Deserialize a PathsReply message. Returns the result for testing.
+        """
+        pfs_reply.deserialize(pfs_reply)
```

Comment:
What's that doing. Isn't the message already deserialized at that point?

Reply:
I assumed messages would arrive in some serialized form at the client, no?

Reply:
I think you can remove this transport stuff from this PR and we can have a call on requirements and expectations to have a better idea what should the mocked client do with the transport. In my tests I send client-generated messages using `DummyTransport` and it works and is received by the server as it should.

https://github.com/raiden-network/raiden-monitoring-service/blob/master/monitoring_service/test/test_monitoring_request.py

When testing matrix transport, a some sort of proxy for client's transport will be required, but the transport itself might not be a mocked client's dependency at all or it might be optional.

Reply:
I'm building it the other way around, I want to send PFS generated messages to the client via 'DummyTransport' which is not really hard I think. The other direction is already working.

Reply:
I'd use gevent's queue to block sync the received events. Something like this (pseudocode)

```
class MockedClient:
  def send_message(self, message):
      assert self.transport is not None
      self.transport.send_message(message)
      if isinstance(message, MessageWeWantToSync):
          reply = self.message_queue.get(timeout=10)  # this blocks until transport receives the reply
          # .... verify reply ...

  def on_message_event(self, message):
    # do some filtering etc
   self.message_queue.put_nowait(message)    # put all received messages to the queue
```